### PR TITLE
fix(datasource/metadata): delete homepage url case

### LINKED ISSUE #14820: Missing source directory for bitnami helm charts
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Some time a go i've implemented a better helm source url handling which works as expected.
But for some reason renovate ignores the `sourceDirectory` property for the links in the PR update table.

- https://github.com/renovate-reproductions/helm-chart-source-url/pull/1

### Relevant debug logs

Nothing relevant found there
<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Changes
`homepage` property should only be deleted if `sourceUrl` has been set to it.
Delete `homepage` if its same as `sourceUrl` after "massaging" it and its not a link to a page.
Added regex to clean preceding slashes for url joining in `pr/body`.
Fixed a wrong test which was passing when `sourceUrl` was having double slashes, and modified a test to have a `homepage` property.

## Context

#Closes https://github.com/renovatebot/renovate/issues/14820

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by PhilipAbed** [comment]
@viceice this is Hasan's task, i fixed your comments, let me know if there is anything you want me to do more here! thanks!

**Review by viceice — COMMENTED** [review]
still needs lint fix, otherwise LGTM

File: lib/modules/datasource/metadata.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ import parse from 'github-url-from-git';
 import { DateTime } from 'luxon';
 import * as hostRules from '../../util/host-rules';
 import { regEx } from '../../util/regex';
-import { parseUrl, validateUrl } from '../../util/url';
+import { parseUrl, trimTrailingSlash, validateUrl } from '../../util/url';
+import { detectPlatform } from '../platform/util';
```
Comment (viceice) [inline lib/modules/datasource/metadata.ts:8]:
```suggestion
import { detectPlatform } from '../../util/common';
```

File: lib/modules/datasource/metadata.ts [hunk 2]
```diff
@@ -4,7 +4,8 @@ import parse from 'github-url-from-git';
 import { DateTime } from 'luxon';
 import * as hostRules from '../../util/host-rules';
 import { regEx } from '../../util/regex';
-import { parseUrl, validateUrl } from '../../util/url';
+import { parseUrl, trimTrailingSlash, validateUrl } from '../../util/url';
+import { detectPlatform } from '../../util/common';
```
Comment (viceice) [inline lib/modules/datasource/metadata.ts:8]: probably needs import reorder

Comment (PhilipAbed) [reply lib/modules/datasource/metadata.ts:8]: its weird, the lint Passed, then i  Updated branch from main, suddenly i get this , ill run lint fix again i guess.

Comment (PhilipAbed) [reply lib/modules/datasource/metadata.ts:8]: someone changed the classes, there's a conflict im fixing

**Review by viceice — APPROVED** [review]

**Pull request merged**